src/Stryker.Core/Stryker.Core/Helpers/TypeBasedStrategy.cs

```diff
@@ -33,23 +33,27 @@ public THandler FindHandler(T item)
 
         private THandler FindHandler(T item, Type type)
```

Comment:
To me the previous style of recursion is more readable. @richardwerkman what do you think

Reply:
Yeah to me "traditional" recursion is more readable. I'm not sure about possible performance gains this way?

Reply:
I did not make any benchmarck to confirm my intuition, but I am concerned since this method is called for every syntax node of the project, and the SyntaxNode inheritance tree is 6 or 7 level deep. Transforming tail recursion to loop is a classic optimisation which typically bring a twofold performance increase for simple function, which is pretty much the case here as we simply try again with the parent class when failing to find a specialised handler.
Nonetheless, it is clear that this loop version was crude, so I implemented a proper for loop that looks far better.

If you still prefer a recursive one, I will revert to it, but feedback on the new version will be welcomed

Reply:
Makes sense, we don't want any stackoverflows.